uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/BAO/Task.php

```diff
@@ -63,4 +63,115 @@ private static function getPreviousAssigneeId($taskId) {
 
     return $id;
   }
+
+  /**
+   * Does user has sufficient permission for view/edit activity record.
+   *
+   * @param int $activityId
+   *   Activity record id.
+   * @param int $action
+   *   Edit/view.
+   *
+   * @return bool
+   */
+  public static function checkPermission($activityId, $action) {
```

Comment:
this is a big method .. it should be separated into more smaller ones to make it more readable